### PR TITLE
[css-grid] Parsing grid-template-rows/columns

### PR DESCRIPTION
**Pull request by ewilligers**
Test parsing of valid and invalid values for
grid-template-rows and grid-template-columns
Value:	`none` | `<track-list>` | `<auto-track-list>`
https://drafts.csswg.org/css-grid/#track-sizing

### CODE REVIEW

**Comment by ewilligers** [comment]
Edge 18 fails by serializing
`"calc(30% + 40vw)"`
as
`"calc(40vw + 30%)"`

Firefox fails by serializing
`"[one] repeat(2, minmax(10px, auto)) [two] 30px [three] repeat(auto-fill, 10px) 40px [four five] repeat(2, minmax(10px, auto)) [six]"`
as
`"[one] repeat(2, minmax(10px, auto)) [two] 30px [three] 40px repeat(auto-fill, 10px) [four five] repeat(2, minmax(10px, auto)) [six]"`

Safari fails by serializing
`"repeat(2, fit-content(20%) [three four] 30px 40px [five six])"`
as
`"fit-content(20%) [three four] 30px 40px [five six] fit-content(20%) [three four] 30px 40px [five six]"`
(Blink does a similar expansion for computed values, not tested in this PR.)

**Comment by ewilligers** [comment]
The other difference in behavior between browsers is discussed in https://github.com/w3c/csswg-drafts/issues/4173 .  This PR accepts each browser's behavior until that question is resolved.

**Comment by ewilligers** [comment]
@tabatkins please review

**Comment by ewilligers** [comment]
@mrego Please review

**Comment by ewilligers** [comment]
@emilio please review

File: css/css-grid/parsing/grid-template-columns-valid.html [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+// <track-repeat> = repeat( [ <positive-integer> ] , [ <line-names>? <track-size> ]+ <line-names>? )
+
+// 'repeat(1, [] 10px)' in Blink
+// 'repeat(1, 10px)' in Firefox
+// '[] 10px' in Safari
+// '10px' in Edge 18
+test_valid_value("grid-template-columns", 'repeat(1, [] 10px)', ['repeat(1, 10px)', 'repeat(1, [] 10px)', '[] 10px', '10px']);
+
+// 'repeat(1, [one two] 20%)' in Blink, Firefox
```
Comment (emilio) [inline css/css-grid/parsing/grid-template-columns-valid.html]: I think Firefox and Blink are right on this one, per https://github.com/w3c/csswg-drafts/issues/3503.

**Review by emilio — APPROVED** [review]
LGTM. It may be worth testing the values in a `display: grid` container, given these have magic resolved-value behavior. But this is probably out-of-scope for this patch and best done (if not covered already, which I _think_ it is) in a separate PR.

File: css/css-grid/parsing/grid-template-columns-valid.html (hunk 1 above)
Comment (ewilligers) [reply css/css-grid/parsing/grid-template-columns-valid.html]: I have made the expectations stricter, to require preservation of `repeat(1, ...)`.

**Comment by ewilligers** [comment]
No need for a `display: grid` container in this test of specified values. It will be relevant for resolved values when we test the values `getComputedStyle()` returns.

**Pull request merged**